### PR TITLE
Add cypress framework

### PR DESCRIPTION
**Pull request by Aboullos**
## Motivation
Cover testing

## Brief Description
Add testing framework and a few tests

## Verification Steps
1- Run KC and the new admin console
2- Go to the folder tests
3- Run yarn
4- Run `npx cypress open` to open Cypress interface, or `npx cypress run` to run the tests

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Formatting has been performed via prettier/eslint
- [ ] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

**Review by edewit — CHANGES_REQUESTED** [review]

**Comment by edewit** [comment]
@Aboullos would it be a good idea to put this into the github action as well?

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
@Aboullos thank you for adding these changes - it's very exciting that we'll be using Cypress as our integration testing framework! 

After navigating to the `/test` directory, the Cypress interface opens on `npx cypress open` and I am able to run the tests. 

In Patternfly we just run `yarn start:cypress` from the root and the interface opens. Would be great if we could do something like `"start:cypress": "npx cypress open",` in the root package.json here as well! Is that possible @Aboullos ?

Also it looks like `client_scopes_test.spec.js` and `login_test.spec.js` currently are not passing

File: tests/.gitignore [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# OS stuff #
+############
+.DS_Store
+
+# Cypress #
+###########
+assets
+cypress.env.json
+
```
Comment (ssilvert) [inline tests/.gitignore:9]: Please remove this file and add what you need to the root .gitignore

We only want a .gitignore here if we need to override what is in the root.

Perhaps all you need is the Cypress stuff?

File: tests/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (ssilvert) [inline tests/LICENSE:1]: Please remove this file.  It already exists in the root.

**Review by ssilvert — CHANGES_REQUESTED** [review]
@Aboullos I'm happy to see this.

Agree with @edewit that we need this to run as a GitHub action.  That is top priority and I don't want to merge until we have that working.

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+* `npm package manager` has to be [downloaded](https://nodejs.org/en/download/) and installed.
+
+### via Cypress Test Runner
+
+**By using `npx`:**
+
+**note**: [npx](https://www.npmjs.com/package/npx) is included with `npm > v5.2` or can be installed separately.
+
+```shell
+npx cypress open
```
[the remark this answers is not in the document]
Comment (Aboullos) [reply tests/README.md:18]: Done, will add it in the next commit squash

File: tests/package.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "cypress-keycloak-ui-testsuite",
+  "version": "1.0.0",
+  "description": "UI tests for Keycloak using Cypress.",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
[the remark this answers is not in the document]
Comment (Aboullos) [reply tests/package.json:7]: Done, will add it in the next commit squash

**Comment by Aboullos** [comment]
> @Aboullos thank you for adding these changes - it's very exciting that we'll be using Cypress as our integration testing framework!
> 
> After navigating to the `/test` directory, the Cypress interface opens on `npx cypress open` and I am able to run the tests.
> 
> In Patternfly we just run `yarn start:cypress` from the root and the interface opens. Would be great if we could do something like `"start:cypress": "npx cypress open",` in the root package.json here as well! Is that possible @Aboullos ?
> 
> Also it looks like `client_scopes_test.spec.js` and `login_test.spec.js` currently are not passing

The login test could be failing if there is no "admin" user with "admin" as password, the client_scope test is failing because the delete a client scope functionality was not working. I did the changes for the "start:cypress", will add it on the next commit.

File: tests/LICENSE (hunk 2 above)
Comment (Aboullos) [reply tests/LICENSE:1]: Done, will add it in the next commit squash

File: tests/.gitignore (hunk 1 above)
Comment (Aboullos) [reply tests/.gitignore:9]: Done, will add it in the next commit squash

**Comment by Aboullos** [comment]
Will check on the GitHub action

**Comment by ssilvert** [comment]
@Aboullos FYI. You don't have to squash your commits on this project.  We just do "Squash and merge" when we merge the PR.

**Comment by ssilvert** [comment]
@Aboullos Very glad to see this running with GitHub Action.  Please either fix or remove the failing test.  We can give this one extra day to merge but I don't want to wait any longer than Tuesday.

**Comment by edewit** [comment]
@ssilvert the test is already fixed PR for it is open #276

**Pull request merged**